gourmet/gtk_extras/TextBufferMarkup.py

```diff
@@ -78,26 +80,35 @@ def __init__ (self):
         GObject.GObject.__init__(self)
 
     def set_text (self, txt):
+        if isinstance(txt, bytes):
+            # data loaded from the database are bytes, not str
+            txt = txt.decode()
         Gtk.TextBuffer.set_text(self,"")
         try:
-            self.parsed,self.txt,self.separator = Pango.parse_markup(txt,'\x00')
-        except:
-            print('Problem encountered escaping text: "%s"'%txt)
+            self.parsed, self.attrList, self.txt, self.separator = Pango.parse_markup(txt, -1, '\x00')
+        except Exception as e:  # unescaped text (g-markup-error-quark), eg. contains &amp
+            print(f"Problem encountered escaping text: {txt}: {e}")
             import traceback; traceback.print_exc()
             txt=xml.sax.saxutils.escape(txt)
-            self.parsed,self.txt,self.separator = Pango.parse_markup(txt,'\x00')
-        self.attrIter = self.parsed.get_iterator()
+            self.parsed, self.attrList, self.txt, self.separator = Pango.parse_markup(txt, -1, '\x00')
+
+        self.attrIter = self.attrList.get_iterator()
         self.add_iter_to_buffer()
-        while next(self.attrIter):
+        while self.attrIter.next():
             self.add_iter_to_buffer()
 
     def add_iter_to_buffer (self):
```

Comment:
Cann we make a static method out of it and give attrIter as a parameter? Then we can destroy it without worries

Reply:
Makes sense, I will do so later today.

Reply:
I've cleaned and documented what's happening here, but it can't really be made a staticmethod.